src/story/story.controller.ts

```
@@ -587,19 +592,26 @@ export class StoryController {
   @ApiResponse({ status: 400, description: 'Bad Request', type: ErrorResponseDto })
   @ApiResponse({ status: 401, description: 'Unauthorized', type: ErrorResponseDto })
+  @ApiResponse({ status: 403, description: 'Story limit reached', type: ErrorResponseDto })
   @ApiResponse({ status: 404, description: 'Not Found', type: ErrorResponseDto })
   async setUserProgress(
-    @Req() req: AuthenticatedRequest,
+    @Req() req: RequestWithStoryAccess,
     @Body() body: UserStoryProgressDto,
   ) {
-    return this.storyService.setUserProgress(req.authUserData.userId, body);
+    // Record access if this is a new story for the user
+    if (req.authUserData?.userId && req.storyAccessResult?.reason !== 'already_read') {
```

ESLint

[error] Replace `req.authUserData?.userId·&&·req.storyAccessResult?.reason·!==·'already_read'` with `⏎······req.authUserData?.userId·&&⏎······req.storyAccessResult?.reason·!==·'already_read'⏎····`

(prettier/prettier)